mobile/src/main/java/org/openhab/habdroid/ui/homescreenwidget/ItemUpdateWidget.kt

```diff
@@ -0,0 +1,237 @@
+import kotlinx.coroutines.launch
+import org.openhab.habdroid.R
+import org.openhab.habdroid.background.BackgroundIntentReceiveActivity
+import org.openhab.habdroid.background.BackgroundTasksManager
+import org.openhab.habdroid.core.connection.ConnectionFactory
+import org.openhab.habdroid.ui.AbstractItemPickerActivity
+import org.openhab.habdroid.ui.PreferencesActivity
+import org.openhab.habdroid.util.HttpClient
+import org.openhab.habdroid.util.bitmapToSvg
+import org.openhab.habdroid.util.dpToPixel
+import org.openhab.habdroid.util.getIconFormat
+import org.openhab.habdroid.util.getPrefs
+import org.openhab.habdroid.util.getString
+import org.openhab.habdroid.util.isIconFormatPng
+import java.io.IOException
+import kotlin.math.min
+
+open class ItemUpdateWidget : AppWidgetProvider() {
+    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+        Log.d(TAG, "onUpdate()")
+        appWidgetIds.forEach {
+            Log.d(TAG, "id: $it")
+            val data = getInfoForWidget(context, it)
+            if (data != null) {
+                setupWidget(context, data, it, appWidgetManager)
+            }
+        }
+        super.onUpdate(context, appWidgetManager, appWidgetIds)
+    }
+
+    override fun onAppWidgetOptionsChanged(
+        context: Context?,
+        appWidgetManager: AppWidgetManager?,
+        appWidgetId: Int,
+        newOptions: Bundle?
+    ) {
+        Log.d(TAG, "onAppWidgetOptionsChanged()")
+        val data = getInfoForWidget(context ?: return, appWidgetId)
+        setupWidget(context, data ?: return, appWidgetId, appWidgetManager ?: return)
+    }
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        Log.d(TAG, "onReceive() $intent")
+        super.onReceive(context, intent)
+    }
+
+    override fun onDeleted(context: Context?, appWidgetIds: IntArray?) {
+        Log.d(TAG, "onDeleted()")
+        if (context == null || appWidgetIds == null) {
+            return
+        }
+        appWidgetIds.forEach {
+            Log.d(TAG, "Deleting data for id $it")
+            context.deleteFile(getFileNameForWidget(it))
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                context.deleteSharedPreferences(getFileNameForWidget(it))
+            } else {
+                saveInfoForWidget(context, ItemUpdateWidgetData("", "", "", "", ""), it)
+            }
+        }
+        super.onDeleted(context, appWidgetIds)
+    }
+
+    companion object {
+        private val TAG = ItemUpdateWidget::class.java.simpleName
+
+        fun setupWidget(
+            context: Context,
+            data: ItemUpdateWidgetData,
+            appWidgetId: Int,
+            appWidgetManager: AppWidgetManager
+        ) {
+            val widgetOptions = appWidgetManager.getAppWidgetOptions(appWidgetId)
+            val smallWidget = widgetOptions.getInt(AppWidgetManager.OPTION_APPWIDGET_MAX_HEIGHT) < 200
```

Comment:
Maybe use a dimension value instead of the magic number '200'? `R.dimen.small_widget_threshold` or something?